File: src/main/java/framboos/OutPin.java
package framboos;

import static framboos.FilePaths.*;

public class OutPin extends GpioPin {
	
	public OutPin(int pinNumber) {
		super(pinNumber, Direction.OUT);
		setValue(false);
	}

	public OutPin(String pinName) {
		super(getPinNumber(pinName), Direction.OUT, true);
		setValue(false);
	}
	
	public void setValue(boolean isOne) {
		if (!isClosing) {
			writeFile(getValuePath(pinNumber), isOne ? "1" : "0");
		}
	}
	
	@Override
	public void close() {
		setValue(false);
		super.close();
	}
}

File: src/main/scala/framboos/OutPin.scala
package framboos;

import GpioPin._

class OutPin(pinNumber: Int, isDirect: Boolean) extends GpioPin(pinNumber, Out, isDirect) {
  def this(pinNumber: Int) = this(mappedPins(pinNumber), isDirect = false)
  def this(pinName: String) = this(getPinNumber(pinName), isDirect = true)

  // initially set pin to off
  setValue(false)

  def setValue(isOne: Boolean) {
    if (!isClosing) {
      writeFile(valuePath(pinNumber), if (isOne) "1" else "0")
    }
  }

  override def close {
    setValue(false)
    super.close
  }
}

object OutPin {
  def apply(pinNumber: Int) = new OutPin(pinNumber)
}
